client/components/social-buttons/google.js

```diff
@@ -139,6 +181,38 @@ class GoogleLoginButton extends Component {
 		return this.initialized;
 	}
 
+	async handleAuthorizationCode( auth_code ) {
+		let response;
+
+		try {
+			response = await postLoginRequest( 'exchange-social-auth-code', {
+				service: 'google',
+				auth_code,
+				client_id: config( 'wpcom_signup_id' ),
+				client_secret: config( 'wpcom_signup_key' ),
+			} );
+		} catch ( httpError ) {
+			const { code: error_code } = getErrorFromHTTPError( httpError );
+
+			if ( error_code ) {
+				this.props.recordTracksEvent( 'calypso_login_auth_code_exchange_failure', {
+					social_account_type: 'google',
+					error_code,
+				} );
+			}
+
+			this.setState( {
+				error: this.props.translate( 'An error has occurred. Please try again.' ),
+			} );
```

Comment:
We cannot set `this.state.error` here because that'd disable the Google button and the user won't be able to try again. We need to dispatch a toast, global notification.